src/server/metrics.rs

```diff
@@ -100,4 +144,78 @@ mod tests {
         //experiment is executed by a subset of the agents
         metrics.on_complete_experiment(ex2).unwrap();
     }
+
+    fn supposed_to_work<F>(metric: &MetricFamily, metric_filter: F) -> bool
```

Comment:
Nit: as you only use `metric_filter` to filter a single agent, you can do `filter_agent: Option<&str>` instead of passing the whole closure.